python/pyspark/sql/dataframe.py

```diff
@@ -1307,43 +1307,66 @@ def replace(self, to_replace, value, subset=None):
         |null|  null|null|
         +----+------+----+
         """
-        if not isinstance(to_replace, (float, int, long, basestring, list, tuple, dict)):
+        # Helper functions
+        def all_of(types):
```

Comment:
Maybe give this a doc-string to clarify what all_of does even though its not user facing better to have a docstring than not.